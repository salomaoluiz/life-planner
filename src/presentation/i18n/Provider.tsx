import React from "react";

import { ReactI18NPProvider } from "./react-i18n";

interface Props {
  children: React.ReactNode;
}

function I18NProvider({ children }: Props) {
  return <ReactI18NPProvider>{children}</ReactI18NPProvider>;
}

export default I18NProvider;
