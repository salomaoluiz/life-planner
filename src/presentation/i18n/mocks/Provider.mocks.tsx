import { View } from "react-native";

import { render } from "@tests";

import { I18NProvider } from "../";
import * as reactI18N from "../react-i18n";

// region Mocks

jest.mock("@presentation/i18n/react-i18n");
jest.unmock("@presentation/i18n");

// endregion Mocks

// region Spies

jest
  .spyOn(reactI18N, "ReactI18NPProvider")
  .mockImplementation(({ children }) => (
    <View testID="react-i18n-provider">{children}</View>
  ));
// endregion Spies

function setup() {
  render(I18NProvider({ children: <View testID="i18n-provider-children" /> }));
}

const spies = {};

export { setup, spies };
export { screen } from "@tests";
