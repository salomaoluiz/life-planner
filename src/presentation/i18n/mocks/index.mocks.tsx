import { View } from "react-native";

import { render, renderHook } from "@tests";

import {
  I18NProvider,
  translate,
  useTranslation,
  useTranslationLocale,
} from "@presentation/i18n";
import * as reactI18N from "@presentation/i18n/react-i18n";
import { reactI18NHooks } from "@presentation/i18n/react-i18n";
import { TranslationKeys } from "@presentation/i18n/types";

jest.mock("@presentation/i18n/react-i18n");
jest.unmock("@presentation/i18n");

const reactI18NUseTranslationSpy = jest.spyOn(reactI18NHooks, "useTranslation");
const reactI18NUseTranslationLocaleSpy = jest.spyOn(
  reactI18NHooks,
  "useTranslationLocale",
);
const reactI18NTranslateSpy = jest.spyOn(reactI18N, "reactI18NTranslate");

jest
  .spyOn(reactI18N, "ReactI18NPProvider")
  .mockImplementation(({ children }) => (
    <View testID="react-i18n-provider">{children}</View>
  ));

function setupProvider() {
  render(I18NProvider({ children: <View testID="i18n-provider-children" /> }));
}

function setupTranslate(key: TranslationKeys, params?: Record<string, string>) {
  return translate(key, params);
}

function setupUseTranslation() {
  return renderHook(() => useTranslation());
}

function setupUseTranslationLocale() {
  return renderHook(() => useTranslationLocale());
}

const setup = {
  provider: setupProvider,
  translate: setupTranslate,
  useTranslation: setupUseTranslation,
  useTranslationLocale: setupUseTranslationLocale,
};

const spies = {
  reactI18NTranslate: reactI18NTranslateSpy,
  reactI18NUseTranslation: reactI18NUseTranslationSpy,
  reactI18NUseTranslationLocale: reactI18NUseTranslationLocaleSpy,
};

export { setup, spies };
export { screen } from "@tests";
