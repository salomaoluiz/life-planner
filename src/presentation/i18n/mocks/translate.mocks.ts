import { translate } from "../";
import * as reactI18N from "../react-i18n";
import { TranslationKeys } from "../types";

// region Mocks

jest.mock("@presentation/i18n/react-i18n");
jest.unmock("@presentation/i18n");

// endregion Mocks

// region Spies
const reactI18NTranslateSpy = jest.spyOn(reactI18N, "reactI18NTranslate");
// endregion Spies

function setup(key: TranslationKeys, params?: Record<string, string>) {
  return translate(key, params);
}

const spies = {
  reactI18NTranslate: reactI18NTranslateSpy,
};

export { setup, spies };
export { screen } from "@tests";
