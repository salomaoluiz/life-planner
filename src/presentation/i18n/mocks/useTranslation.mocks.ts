import { renderHook } from "@tests";

import { useTranslation } from "../";
import { reactI18NHooks } from "../react-i18n";

// region Mocks

jest.mock("@presentation/i18n/react-i18n");
jest.unmock("@presentation/i18n");

// endregion Mocks

// region Spies
const reactI18NUseTranslationSpy = jest.spyOn(reactI18NHooks, "useTranslation");
// endregion Spies

function setupUseTranslation() {
  return renderHook(() => useTranslation());
}

const setup = {
  useTranslation: setupUseTranslation,
};

const spies = {
  reactI18NUseTranslation: reactI18NUseTranslationSpy,
};

export { setup, spies };
export { screen } from "@tests";
