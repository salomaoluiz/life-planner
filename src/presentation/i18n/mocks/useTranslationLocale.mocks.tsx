import { renderHook } from "@tests";

import { useCases } from "@application/useCases";
import * as fetcher from "@infrastructure/fetcher";
import UseMutationFixture from "@infrastructure/fetcher/mocks/useMutation.fixture";

import { useTranslationLocale } from "../";
import { reactI18NHooks } from "../react-i18n";
import { availableLanguages } from "../translations";

// region Mocks

jest.mock("@presentation/i18n/react-i18n");
jest.unmock("@presentation/i18n");
jest.mock("@infrastructure/fetcher");

const useMutationFixture = new UseMutationFixture().reset();
const useMutationResponse = useMutationFixture.build();
const getLocaleResponse = {
  languageTag: "en",
};

const i18NUseTranslationLocaleResponse = {
  changeLocale: jest.fn(),
  getLocale: jest.fn().mockReturnValue(getLocaleResponse),
};
// endregion Mocks

// region Spies
const reactI18NUseTranslationLocaleSpy = jest
  .spyOn(reactI18NHooks, "useTranslationLocale")
  .mockReturnValue(i18NUseTranslationLocaleResponse);

const useMutationSpy = jest
  .spyOn(fetcher, "useMutation")
  .mockReturnValue(useMutationResponse);
// endregion Spies

function setup() {
  return renderHook(() => useTranslationLocale());
}

const spies = {
  reactI18NUseTranslationLocale: reactI18NUseTranslationLocaleSpy,
  useMutation: useMutationSpy,
};

const mocks = {
  availableLanguages,
  getLocaleResponse,
  i18NUseTranslationLocaleResponse,
  useCases,
  useMutationResponse,
};

beforeEach(() => {
  jest.clearAllMocks();
});

export { mocks, setup, spies };
export { act, screen } from "@tests";
