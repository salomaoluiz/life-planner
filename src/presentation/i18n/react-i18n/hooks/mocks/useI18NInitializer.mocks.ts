import i18next from "i18next";
import { initReactI18next } from "react-i18next";

import { renderHook } from "@tests";

import ConfigsDTO from "@application/dto/configs/ConfigsDTO";
import * as fetcher from "@infrastructure/fetcher";
import UseQueryFixture from "@infrastructure/fetcher/mocks/useQuery.fixture";
import { translations } from "@presentation/i18n/translations";
import * as loaderProvider from "@providers/loader";

import useI18NInitializer from "../useI18NInitializer";

// region Mocks
jest.mock("i18next", () => {
  const originalModule = jest.requireActual("i18next");
  return {
    ...originalModule,
    language: "en-US",
    use: jest.fn().mockReturnValue({
      init: jest.fn(),
    }),
  };
});
jest.mock("@infrastructure/fetcher");

const loaderProviderResponse = {
  isLoading: false,
  setIsLoading: jest.fn(),
};
const useQueryFixture = new UseQueryFixture<Partial<ConfigsDTO>>().reset();
const useQueryPendingResponse = useQueryFixture.build();
const useQuerySuccessFixture = useQueryFixture.withStatus("success");

const useQuerySuccessWithDataResponse = useQuerySuccessFixture
  .withData({
    language: "en-US",
  })
  .build();
const useQuerySuccessResponse = useQuerySuccessFixture.build();

// endregion Mocks

// region Spies
const initSpy = jest.fn().mockResolvedValue({});
const useSpy = jest.spyOn(i18next, "use").mockReturnValue({
  init: initSpy,
} as never);

jest
  .spyOn(loaderProvider, "useProviderLoader")
  .mockReturnValue(loaderProviderResponse);
const useQuerySpy = jest
  .spyOn(fetcher, "useQuery")
  .mockReturnValue(useQueryPendingResponse);
// region Spies

function setup() {
  return renderHook(() => useI18NInitializer());
}

const spies = {
  init: initSpy,
  initReactI18next,
  use: useSpy,
  useQuery: useQuerySpy,
};

const mocks = {
  i18next,
  loaderProviderResponse,
  translations,
  useQuery: {
    pending: {
      response: useQueryPendingResponse,
    },
    success: {
      empty: useQuerySuccessResponse,
      withData: useQuerySuccessWithDataResponse,
    },
  },
};

beforeEach(() => {
  jest.clearAllMocks();
});

export { mocks, setup, spies };

export { waitFor } from "@tests";
