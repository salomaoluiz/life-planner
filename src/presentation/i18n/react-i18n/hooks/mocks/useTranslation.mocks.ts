import * as reactI18Next from "react-i18next";

import { renderHook } from "@tests";

import useTranslation from "../useTranslation";

// region Mocks
jest.mock("react-i18next");

// endregion Mocks

// region Spies
const tSpy = jest.fn();

jest
  .spyOn(reactI18Next, "useTranslation")
  .mockReturnValue({ t: tSpy } as never);

// endregion Spies

function setup() {
  return renderHook(() => useTranslation());
}

const spies = {
  t: tSpy,
};

beforeEach(() => {
  jest.clearAllMocks();
});

export { setup, spies };
export { act } from "@tests";
