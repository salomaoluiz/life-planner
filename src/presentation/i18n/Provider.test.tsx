import { screen, setup } from "./mocks/Provider.mocks";

it("SHOULD render the react-i18n provider", () => {
  setup();

  expect(screen.getByTestId("react-i18n-provider")).toBeOnTheScreen();
  expect(screen.getByTestId("i18n-provider-children")).toBeOnTheScreen();
  expect(screen.toJSON()).toMatchSnapshot();
});
