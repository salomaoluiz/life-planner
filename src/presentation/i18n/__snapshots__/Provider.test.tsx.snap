// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHOULD render the react-i18n provider 1`] = `
<View
  testID="react-i18n-provider"
>
  <View
    testID="i18n-provider-children"
  />
</View>
`;
